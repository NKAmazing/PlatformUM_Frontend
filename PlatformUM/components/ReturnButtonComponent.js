import React from "react";
import { TouchableOpacity, Text } from "react-native";
import { Image } from "react-native";

const ReturnButtonComponent = () => {
    return (
        <TouchableOpacity
            style={styles.returnButton}
            onPress={() => {
                // Here is the navigation to the previous screen
            }}
        >
           <Image
                source={require('../assets/return-button-icon.png')}
                style={styles.icon}
            /> 
        </TouchableOpacity>
    );
}

const styles = {
    returnButton: {
        width: 50,  // Width of the button
        height: 50, // Height of the button
        alignItems: 'center',
        justifyContent: 'center',
        marginLeft: 5, // Margin left of the button to align position
        // backgroundColor: 'black',
        borderRadius: 30, // Border radius of the button
    },
    // buttonText: {
    //     fontSize: 16,
    //     color: 'white',
    // },
    icon: {
        width: 30, // Width of the icon
        height: 30, // Height of the icon
    },
};


export default ReturnButtonComponent;