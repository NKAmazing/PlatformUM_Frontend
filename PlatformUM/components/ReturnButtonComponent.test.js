import React from 'react';
import renderer from 'react-test-renderer';
import ReturnButtonComponent from './ReturnButtonComponent';

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ goBack: jest.fn() }),
}));

test('ReturnButtonComponent renders correctly', () => {
  const tree = renderer.create(<ReturnButtonComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});