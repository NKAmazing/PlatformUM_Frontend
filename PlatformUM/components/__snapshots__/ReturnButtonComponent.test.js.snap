// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReturnButtonComponent renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 30,
      "height": 50,
      "justifyContent": "center",
      "marginLeft": 5,
      "opacity": 1,
      "width": 50,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 30,
        "width": 30,
      }
    }
  />
</View>
`;
